src/Proto3/Suite/DotProto/Parsing.hs

```diff
@@ -195,19 +199,24 @@ sortStatements modulePath statements
     adapt _     = DotProtoNoPackage
 
 topLevel :: Path -> ProtoParser DotProto
-topLevel modulePath = do whiteSpace
-                         syntaxSpec
-                         sortStatements modulePath <$> many topStatement
+topLevel modulePath = do
+  whiteSpace
+  syntaxSpec
+  sortStatements modulePath <$> many topStatement
 
 --------------------------------------------------------------------------------
 -- top-level statements
 
 topStatement :: ProtoParser DotProtoStatement
-topStatement = DPSImport     <$> import_
-           <|> DPSPackage    <$> package
-           <|> DPSOption     <$> pOptionStmt
-           <|> DPSDefinition <$> definition
-           <|> DPSEmpty      <$  empty
+topStatement = 
+  choice 
+    [ DPSImport <$> import_
+    , DPSPackage <$> package
+    , DPSOption <$> pOptionStmt
+    , DPSEmpty <$ try pExtendStmt
```

Comment:
Why do we need `try` here but no where else?

Reply:
Ideally I could just have written `try pExtendStmt <|> enum` within the `definition` parser, but because theres no AST to represent message extensions I need to parse extensions at the top-level return `DPSEmpty` then bubble up backtracking the between `enum` and `extend` to this parser.

Reply:
Okay, thanks for explaining!